skimage/data/generate_shapes.py

```diff
@@ -0,0 +1,316 @@
+        [min_pixel_intensity, 255], else an array of three elements, each in
+        the range of [min_pixel_intensity, 255].
+
+    """
+    if not (0 <= min_pixel_intensity <= 255):
+        raise ValueError('Minimum intensity must be in interval [0, 255]')
+    if gray:
+        return random.randint(min_pixel_intensity, 255)
+    return random.randint(min_pixel_intensity, 255, size=3)
+
+
+def generate_shapes(image_shape,
+                    max_shapes,
+                    min_shapes=1,
+                    min_size=2,
+                    max_size=None,
+                    gray=False,
+                    shape=None,
+                    min_pixel_intensity=0,
+                    allow_overlap=False,
+                    number_of_attemps_to_fit_shape=100,
+                    random_seed=None):
+    """Generate an image with random shapes, labeled with bounding boxes.
+
+    The image is populated with random shapes with random sizes, random
+    locations, and random colors, with or without overlap.
+
+    Shapes have random (row, col) starting coordinates and random sizes bounded
+    by `min_size` and `max_size`. It can occur that a randomly generated shape
+    will not fit the image at all. In that case, the algorithm will try again
+    with new starting coordinates a certain number of times. However, it also
+    means that some shapes may be skipped altogether. In that case, this
+    function will generate fewer shapes than requested.
+
+    Parameters
+    ----------
+    image_shape : (int, int)
+        The number of rows and columns of the image to generate.
+    max_shapes : int
+        The maximum number of shapes to (attempt to) fit into the shape.
+    min_shapes : int
+        The minimum number of shapes to (attempt to) fit into the shape.
+    min_size : int
+        The minimum dimension of each shape to fit into the image.
+    max_size : int
+        The maximum dimension of each shape to fit into the image.
+    gray : bool
+        If `True`, generate 1-D monochrome images, else 3-D RGB images.
+    shape : {rectangle, circle, triangle, None} str
+        The name of the shape to generate or `None` to pick random ones.
+    min_pixel_intensity : [0-255] int
+        The minimum pixel value for colors.
+    allow_overlap : bool
+        If `True`, allow shapes to overlap.
+    number_of_attemps_to_fit_shape : int
+        How often to attempt to fit a shape into the image before skipping it.
+    seed : int
+        Seed to initialize the random number generator.
+        If `None`, a random seed from the operating system is used.
+
+    Returns
+    -------
+    image : uint8 array
+        An image with the fitted shapes.
+    labels : list
+        A list of Label namedtuples, one per shape in the image.
+
+    Examples
+    --------
+    >>> import skimage.data
+    >>> image, labels = skimage.data.generate_shapes((32, 32), max_shapes=3)
+    >>> image
+    array([
```

Comment:
doctests are failing here (see Travis log), I guess this is because the result is not reproducible. Try either to fix the seed or remove this line.